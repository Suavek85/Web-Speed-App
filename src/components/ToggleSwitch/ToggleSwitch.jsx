import React, { useState, useEffect } from 'react'
import './ToggleSwitch.scss'

export default function ToggleSwitch() {

  const DARK_CLASS = "dark";
  const [isDark, setIsDark] = useState(false);

  useEffect(() => {
    isDark ? document.documentElement.classList.add(DARK_CLASS) : document.documentElement.classList.remove(DARK_CLASS)   
  }, [isDark]);

  return (
    <div className="theme-switch-wrapper">
        <label className="theme-switch" htmlFor="checkbox">
            <input 
              type="checkbox" 
              id="checkbox" 
              onChange={event => setIsDark(event.target.checked)} 
            />
            <div className="slider round"></div>
      </label>
    </div>
  )
}
 