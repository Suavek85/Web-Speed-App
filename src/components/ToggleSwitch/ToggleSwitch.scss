@import '../../scss//styles';

.theme-switch-wrapper {
  display: flex;
  align-items: center;
  margin-right: 20px;
  transform: scale(0.8);
  em {
    margin-left: 10px;
    font-size: 1rem;
  }
}

.theme-switch {
  position: relative;
  display: inline-block;
  width: 60px;
  height: 34px;
}

.theme-switch input {
  display: none;
}

.slider {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  cursor: pointer;
  background-color: #ccc;
  transition: 0.4s;
}

.slider::before {
  position: absolute;
  bottom: 4px;
  left: 4px;
  width: 26px;
  height: 26px;
  content: '';
  background-color: #fff;
  transition: 0.4s;
}

input:checked + .slider {
  background-color: var(--color-basic-background);
}

input:checked + .slider::before {
  transform: translateX(26px);
}

.slider.round {
  border-radius: 34px;
}

.slider.round::before {
  border-radius: 50%;
}
