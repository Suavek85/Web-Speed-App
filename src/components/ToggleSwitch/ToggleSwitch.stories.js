import React from 'react'
import { storiesOf } from '@storybook/react'
import { withKnobs, select } from "@storybook/addon-knobs"
import ToggleSwitch from './ToggleSwitch.jsx'

const stories = storiesOf('Toggle', module)
stories.addDecorator(withKnobs)

stories.add('Toggle Switch', () => {
  return <ToggleSwitch  />
})