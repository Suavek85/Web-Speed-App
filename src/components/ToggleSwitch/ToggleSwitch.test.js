import React from 'react'
import { shallow } from 'enzyme'
import ToggleSwitch from './ToggleSwitch'

describe('ToggleSwitch snapshot', () => {

  it('should match the snapshot', () => {
    const wrapper = shallow(<ToggleSwitch />)
    expect(wrapper).toMatchSnapshot()
  })

})