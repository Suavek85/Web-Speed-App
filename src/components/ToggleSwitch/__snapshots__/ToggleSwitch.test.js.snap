// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleSwitch snapshot should match the snapshot 1`] = `
<div
  className="theme-switch-wrapper"
>
  <label
    className="theme-switch"
    htmlFor="checkbox"
  >
    <input
      defaultChecked={false}
      id="checkbox"
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="slider round"
    />
  </label>
</div>
`;
